server/core/src/main/java/org/infinispan/server/core/transport/IpSubnetFilterRule.java

```diff
@@ -0,0 +1,189 @@
+package org.infinispan.server.core.transport;
+
+import java.math.BigInteger;
+import java.net.Inet4Address;
+import java.net.Inet6Address;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.UnknownHostException;
+import java.util.Objects;
+
+import io.netty.handler.ipfilter.IpFilterRule;
+import io.netty.handler.ipfilter.IpFilterRuleType;
+import io.netty.util.internal.ObjectUtil;
+import io.netty.util.internal.SocketUtils;
+
+/**
+ * This differs from Netty's equivalent {@link io.netty.handler.ipfilter.IpSubnetFilterRule} in that it parses CIDR
+ * addresses
+ *
+ * @author Tristan Tarrant
+ * @since 12.1
+ */
+public abstract class IpSubnetFilterRule implements IpFilterRule {
```

Comment:
well, I have a "smaller" implementation in mind:
```
public class InfinispanIpSubnetFilterRule implements IpFilterRule {

   private final IpSubnetFilterRule filterRule;
   private final String cidr;

   public InfinispanIpSubnetFilterRule(String cidr, IpFilterRuleType ruleType) {
      int sep = cidr.indexOf('/');
      if (sep < 0) {
         throw new IllegalArgumentException(cidr);
      }
      this.filterRule = new IpSubnetFilterRule(cidr.substring(0, sep), Integer.parseInt(cidr.substring(sep + 1)), ruleType);
      this.cidr = cidr;
   }

   public String cidr() {
      return cidr;
   }

   @Override
   public boolean matches(InetSocketAddress inetSocketAddress) {
      return filterRule.matches(inetSocketAddress);
   }

   @Override
   public IpFilterRuleType ruleType() {
      return filterRule.ruleType();
   }
}
```

Reply:
Not sure how that improves things :)

Reply:
it doesn't. But this code seems copied from netty. Any bugfixes and/or improvements made by them, we won't get them.